### PR TITLE
Move Index to use kclient

### PR DESCRIPTION
**Pull request by howardjohn**
This allows:

* Proper HasSynced support for Index
* Allow things using index to use kclient
* Allow proper use of filtered informer with Index

### CODE REVIEW

File: security/pkg/server/ca/server.go [hunk 1]
```diff
@@ -168,7 +168,7 @@ func (s *Server) Register(grpcServer *grpc.Server) {
 }
 
 // New creates a new instance of `IstioCAServiceServer`
-func New(ca CertificateAuthority, ttl time.Duration, authenticators []security.Authenticator, client kube.Client) (*Server, error) {
+func New(ca CertificateAuthority, ttl time.Duration, authenticators []security.Authenticator, client kube.Client, filter func(t any) bool) (*Server, error) {
```
Comment (hzxuzhonghu) [inline security/pkg/server/ca/server.go:171]: `filter` should be more readable, to filter what?

Comment (howardjohn) [reply security/pkg/server/ca/server.go:171]: Its the discovery filter fwiw but I'll rename the variable to be more clear

Comment (howardjohn) [reply security/pkg/server/ca/server.go:171]: I put a hold if you want to approve and I can rename tomorrow?

Comment (howardjohn) [reply security/pkg/server/ca/server.go]: updated

**Review by stevenctl — APPROVED** [review]

**Review by lei-tang — APPROVED** [review]
LGTM, approved. I added a do-not-merge label so the PR will not be merged before @hzxuzhonghu has a chance to take a look whether his comments are addressed.

Please feel free to remove the do-not-merge label when the PR is ready to be merged.

**Pull request merged**